tck/features/ListOperations.feature

```diff
@@ -0,0 +1,748 @@
+      | false  |
+    And no side effects
+
+  Scenario: Equality between almost equal lists with null should return null
+    When executing query:
+      """
+      RETURN [1, 2] = [null, 2] AS res
+      """
+    Then the result should be:
+      | res   |
+      | null  |
+    And no side effects
+
+  Scenario: Equality of nested lists of different length should return false despite nulls
+    When executing query:
+      """
+      RETURN [[1]] = [[1], [null]] AS res
+      """
+    Then the result should be:
+      | res    |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between different nested lists with null should return false
+    When executing query:
+      """
+      RETURN [[1, 2], [1, 3]] = [[1, 2], [null, 'foo']] AS res
+      """
+    Then the result should be:
+      | res    |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between almost equal nested lists with null should return null
+    When executing query:
+      """
+      RETURN [[1, 2], ['foo', 'bar']] = [[1, 2], [null, 'bar']] AS res
+      """
+    Then the result should be:
+      | res    |
+      | null  |
+    And no side effects
+
+  # General
+
+  Scenario: Return list size
+    When executing query:
+      """
+      RETURN size([1, 2, 3]) AS n
+      """
+    Then the result should be:
+      | n |
+      | 3 |
+    And no side effects
+
+  Scenario: Setting and returning the size of a list property
+    And having executed:
+      """
+      CREATE (:Label)
+      """
+    When executing query:
+      """
+      MATCH (n:Label)
+      SET n.x = [1, 2, 3]
+      RETURN size(n.x)
+      """
+    Then the result should be:
+      | size(n.x) |
+      | 3         |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Concatenating and returning the size of literal lists
+    When executing query:
+      """
+      RETURN size([[], []] + [[]]) AS l
+      """
+    Then the result should be:
+      | l |
+      | 3 |
+    And no side effects
+
+  Scenario: Returning nested expressions based on list property
+    And having executed:
+      """
+      CREATE (:Label)
+      """
+    When executing query:
+      """
+      MATCH (n:Label)
+      SET n.array = [1, 2, 3, 4, 5]
+      RETURN tail(tail(n.array))
+      """
+    Then the result should be:
+      | tail(tail(n.array)) |
+      | [3, 4, 5]           |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Indexing into nested literal lists
+    When executing query:
+      """
+      RETURN [[1]][0][0]
+      """
+    Then the result should be:
+      | [[1]][0][0] |
+      | 1           |
+    And no side effects
+
+  Scenario: Concatenating lists of same type
+    When executing query:
+      """
+      RETURN [1, 10, 100] + [4, 5] AS foo
+      """
+    Then the result should be:
+      | foo                |
+      | [1, 10, 100, 4, 5] |
+    And no side effects
+
+  Scenario: Appending lists of same type
```

Comment:
This scenario calls this `appending` and the above calls it `concatenating`. We should try to be consistent. I vote for `concatenating` because it isn't clear whether the operation is `appending` or `prepending`, which are both "directed".